LoopKitUI/Views/Settings Editors/InsulinModelSelection.swift

```diff
@@ -77,27 +103,44 @@ public struct InsulinModelSelection: View, HorizontalSizeClassOverride {
         return chartManager
     }()
 
-    @Environment(\.dismiss) var dismiss
-    
     public var body: some View {
         switch mode {
-        case .acceptanceFlow: return AnyView(contentWithSaveButton)
-        case .settings:       return AnyView(contentWithSaveButton)
+        case .acceptanceFlow: return AnyView(content)
+        case .settings: return AnyView(contentWithCancel)
         case .legacySettings: return AnyView(navigationContent)
         }
     }
     
+    private var contentWithCancel: some View {
+        if value == initialValue {
+            return AnyView(content
+                .navigationBarBackButtonHidden(false)
+                .navigationBarItems(leading: EmptyView())
+            )
+        } else {
+            return AnyView(content
+                .navigationBarBackButtonHidden(true)
+                .navigationBarItems(leading: cancelButton)
+            )
+        }
+    }
+    
+    private var cancelButton: some View {
+        Button(action: { self.cancel?() } ) { Text("Cancel", comment: "Cancel editing settings button title") }
+    }
+
     private var navigationContent: some View {
         NavigationView {
-            contentWithSaveButton
-            .navigationBarItems(leading: dismissButton)
+            content
+                .navigationBarItems(leading: dismissButton)
+                .navigationBarTitle(Text(TherapySetting.insulinModel.title), displayMode: .large)
```

Comment:
There's no title for the other configurations of this screen, which makes it so no title appears in the acceptance flow or the new therapy settings screen.

Reply:
I was thrown by the fact that sometimes the title "Insulin Model" got "stuck" sometimes when navigating back.  I thought this would fix it, but turns out this is wrong, too.  I can't reproduce it, so I'll put this back to where it used to be.  But I'll keep an eye out for it.